src/views/studio/VideoWorkspace/VideoWorkspaceForm/VideoWorkspaceForm.tsx

```diff
@@ -64,629 +76,877 @@ const knownLicensesOptions: SelectItem<License['code']>[] = knownLicenses.map((l
+          data: VideoWorkspaceFormFields,
+          addDraftFn: typeof addDraft,
+          updateDraftFn: typeof updateDraft,
+          updateSelectedTabFn: typeof updateSelectedVideoTab
+        ) => {
+          const draftData: RawDraft = {
+            ...data,
+            channelId: activeChannelId,
+            type: 'video',
+            publishedBeforeJoystream: isDateValid(data.publishedBeforeJoystream)
+              ? formatISO(data.publishedBeforeJoystream as Date)
+              : null,
+          }
+          if (tab.isNew) {
+            addDraftFn(draftData, tab.id)
+            updateSelectedTabFn({ isNew: false })
+          } else {
+            updateDraftFn(tab.id, draftData)
+          }
+        },
+        700
+      )
+    )
+
+    const debouncedSetSelectedVideoTabCachedDirtyFormData = useRef(
+      debounce(
+        (
+          data: VideoWorkspaceFormFields,
+          dirtyFields: DeepMap<VideoWorkspaceFormFields, true>,
+          setSelectedVideoTabCachedDirtyFormDataFn: typeof setSelectedVideoTabCachedDirtyFormData
+        ) => {
+          const keysToKeep = Object.keys(dirtyFields) as Array<keyof VideoWorkspaceFormFields>
+          const dirtyData = keysToKeep.reduce((acc, curr) => {
+            acc[curr] = data[curr]
+            return acc
+          }, {} as Record<string, unknown>)
+          setSelectedVideoTabCachedDirtyFormDataFn(dirtyData)
+        },
+        700
+      )
+    )
+
+    useEffect(() => {
+      if (tabDataLoading || !tabData || !selectedVideoTab) {
+        return
+      }
 
-  // we pass the functions explicitly so the debounced function doesn't need to change when those functions change
-  const debouncedDraftSave = useRef(
-    debounce(
-      (
-        tab: VideoWorkspaceTab,
+      // only run this hook if the selected tab changed or we forced reset
+      if (selectedVideoTab.id === cachedSelectedVideoTabId && !forceReset) {
+        return
+      }
+      setCachedSelectedVideoTabId(selectedVideoTab.id)
+      setForceReset(false)
+
+      // flush any possible changes to the edited draft
+      debouncedDraftSave.current.flush()
+
+      setFileSelectError(null)
+      reset(tabData)
+
+      if (selectedVideoTabCachedDirtyFormData) {
+        // allow a render for the form to reset first and then set fields dirty
+        setTimeout(() => {
+          const keys = Object.keys(selectedVideoTabCachedDirtyFormData) as Array<keyof VideoWorkspaceFormFields>
+          keys.forEach((key) => {
+            setValue(key, selectedVideoTabCachedDirtyFormData[key] as ValueOf<VideoWorkspaceFormFields>, {
+              shouldDirty: true,
+            })
+          })
+        }, 0)
+      }
+    }, [
+      selectedVideoTab,
+      cachedSelectedVideoTabId,
+      forceReset,
+      reset,
+      tabDataLoading,
+      tabData,
+      updateSelectedVideoTab,
+      selectedVideoTabCachedDirtyFormData,
+      setValue,
+    ])
+
+    const onSubmit = useCallback(
```

Comment:
I can't comment on these lines , but I think we could safely refactor lines 156-159 to: 
```tsx
    const addAsset = useAssetStore((state) => state.actions.addAsset)
    const mediaAsset = useRawAsset(watch('assets.video.contentId'))
    const thumbnailAsset = useRawAsset(watch('assets.thumbnail.cropContentId'))
    const originalThumbnailAsset = useRawAsset(watch('assets.thumbnail.originalContentId'))
```
and if we do that we could remove `defaultValues` from line 145 as well. It seems that we don't need it at all for some reason. Did some testing and I think we should be fine with these changes